src/libcore/iter/adapters/mod.rs

```diff
@@ -1509,6 +1509,15 @@ impl<I> DoubleEndedIterator for Skip<I> where I: DoubleEndedIterator + ExactSize
         }
     }
 
+    #[inline]
+    fn nth_back(&mut self, n: usize) -> Option<I::Item> {
+        if n < self.len() {
+            self.iter.nth_back(n)
+        } else {
+            None
```

Comment:
I guess codes like this will be broken:

```rust
let mut it = vec![1, 2, 3, 4].into_iter().skip(1);
assert!(it.nth_back(3) == None); // => NOP with this PR
assert!(it.nth_back(0) == None); // => `Some(4)`?
```